pkg/git/scm_test.go

```diff
@@ -0,0 +1,27 @@
+package git_test
+
+import (
+	"github.com/jenkins-x/go-scm/scm/factory"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/weaveworks/pctl/pkg/git"
+)
+
+var _ = Describe("scm", func() {
+	When("we are trying to create a pull request", func() {
+		It("can use an scm client to talk to the platform", func() {
+			fakeScm, err := factory.NewClient("fake", "", "")
+			Expect(err).NotTo(HaveOccurred())
+			client, err := git.NewClient(git.SCMConfig{
+				Branch: "",
+				Base:   "",
+				Repo:   "",
+				Client: fakeScm,
+			})
+			Expect(err).NotTo(HaveOccurred())
+			err = client.CreatePullRequest()
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+})
```

Comment:
do we want to assert the client was called correctly? Also missing an error test case :smile:

Reply:
I can't. I'm using a real client there with a fake configuration from the actual SCM implementation. That's why there are no error cases... I can't make the fake client return something nasty. But I'll figure out a different way then. This was convenient to implement. :)

Reply:
Actually, I retract this. I was able to do some error scenarios. :)